timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -5,6 +5,8 @@
 import java.time.ZonedDateTime;
 import java.util.*;
 import java.util.concurrent.*;
+import java.util.concurrent.locks.ReadWriteLock;
```

Comment:
the ReadWriteLock on those class members value change is a temporary solution.
plan to extract those 2 members into a provider class, with access control to its class members

Reply:
Thus, all of the value access to those index fields (besides the tests, )are implemented inside this class. I don't want to expose this concurrent access implementation outside.
probably, after implementing the provider class, most of the access code will be moved to TasksMergerJob